nifi-nar-bundles/nifi-jms-bundle/nifi-jms-processors/src/main/java/org/apache/nifi/jms/processors/AbstractJMSProcessor.java

```diff
@@ -196,7 +207,10 @@ private void buildTargetResource(ProcessContext context) {
 
             this.cachingConnectionFactory = new CachingConnectionFactory(cfCredentialsAdapter);
             this.cachingConnectionFactory.setSessionCacheSize(Integer.parseInt(context.getProperty(SESSION_CACHE_SIZE).getValue()));
-
+            String clientId = context.getProperty(CLIENT_ID).getValue();
```

Comment:
just added evaluateAttributeExpressions() to the call chain. i assume this is what is needed (sorry not a nifi code expert)